lib/mysql.js

```diff
@@ -435,7 +435,10 @@ MySQL.prototype.fromColumnValue = function(prop, val) {
         }
         break;
       case 'Boolean':
-        val = Boolean(val);
+        // BIT(1) case: <Buffer 01> for true and <Buffer 00> for false
+        // CHAR(1) case: '1' for true and '0' for false
+        // TINYINT(1) case: 1 for true and 0 for false
+        val = Buffer.isBuffer(val) ? Boolean(val[0]) : Boolean(parseInt(val));
```

Comment:
Should we act more defensively and check that the buffer has a single byte only, before treating it as a `BIT(1)`?

```suggestion
        val = Buffer.isBuffer(val) && val.length === 1 ? Boolean(val[0]) : Boolean(parseInt(val));
```